pubsub/service.go

```diff
@@ -130,3 +130,14 @@ func newExactlyOnceBackoff() gax.Backoff {
 		Multiplier: 2,
 	}
 }
+
+// parseResourceName parses the project and resource ID from a fully qualified name.
+// For example, "projects/p/topics/my-topic" -> "p", "my-topic"
+func parseResourceName(fqn string) (string, string) {
+	s := strings.Split(fqn, "/")
+	// Some tests don't use FQN, in which case return empty projectID.
+	if len(s) == 1 {
+		return "", s[0]
```

Comment:
Not for tests, since we're instantiating the `iterator` directly via [newMessageIterator](https://github.com/hongalex/google-cloud-go/blob/pubsub-otel-trace-links/pubsub/iterator_test.go#L552) We should probably change the tests to use an actual project. If that's preferred, I can make that change and remove the check here.